src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -640,6 +652,123 @@ class ApiCallV2(apiEnv: ApiEnv) extends SparkSessionWrapper {
     }
   }
 
+  /**
+   * Function to make parallel API calls. Currently this functions supports only SqlQueryHistory and ClusterEvents
+   * @param endpoint
+   * @param jsonInput
+   * @param config
+   * @return
+   */
+  def makeParallelApiCalls(endpoint: String, jsonInput: Map[String, String], config: Config): String = {
+    var startValue = jsonInput.get("start_value").get.toLong
+    val endValue = jsonInput.get("end_value").get.toLong
+    val incrementCounter = jsonInput.get("increment_counter").get.toLong
+    val finalResponseCount = jsonInput.get("final_response_count").get.toLong
```

Comment:
same goes to 
 val incrementCounter = jsonInput.get("increment_counter").get.toLong
 val finalResponseCount = jsonInput.get("final_response_count").get.toLong 
this can cause null pointer exception.